File: Transport-Management-Systems-Portal-Order-Service-REST-API/Models/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Transport_Management_Systems_Portal_Order_Service_REST_API.Models.Enums;

namespace Transport_Management_Systems_Portal_Order_Service_REST_API.Models
{
    public class Order
    {
        [Key]
        [Required]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        [ForeignKey("Client")]
        public Guid ClientId { get; set; } = Guid.NewGuid();

        public Client Client { get; set; } = default!;

        [Required]
        public string OrderNumber { get; set; } = default!;

        [Required]
        public OrderStatus Status { get; set; } = OrderStatus.Created;

        [Required]
        public string Priority { get; set; } = string.Empty;

        [Required]
        [ForeignKey("ShipmentAddressId")]
        public Guid ShipmentAddressId { get; set; } = Guid.NewGuid();

        public Address ShipmentAddress { get; set; } = default!;

        [Required]
        [ForeignKey("DeliveryAddressId")]
        public Guid DeliveryAddressId { get; set; } = Guid.NewGuid();

        public Address DeliveryAddress { get; set; } = default!;

        public ICollection<Shipment> Shipments { get; set; } = new List<Shipment>();

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime UpdatedAt { get; set; }
    }
}

File: Transportation-Management-Systems-Document-Service-REST-API/src/main/kotlin/com/example/tms/document_service_rest_api/models/Order.kt
package com.example.tms.document_service_rest_api.models

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.persistence.Column
import java.util.Date
import java.util.UUID

@Entity(name = "Orders")
@Table(name = "Orders", schema = "tms")
data class Order(
    @Id
    @Column(name = "Id")
    val id: UUID? = null,
    @Column(name = "ClientId")
    val clientId: UUID,
    @Column(name = "OrderNumber")
    val orderNumber: String,
    @Column(name = "Status")
    val status: Int,
    @Column(name = "Priority")
    val priority: String,
    @Column(name = "ShipmentAddressId")
    val shipmentAddressId: UUID,
    @Column(name = "DeliveryAddressId")
    val deliveryAddressId: UUID,
    @Column(name = "CreatedAt")
    val createdAt: Date,
    @Column(name = "UpdatedAt")
    val updatedAt: Date,
)
